doc/modules/clustering.rst

```diff
@@ -746,23 +746,15 @@ enable only merging of neighboring pixels on an image, as in the
     :func:`sklearn.neighbors.kneighbors_graph`. In the limit of a small
     number of clusters, they tend to give a few macroscopically occupied
     clusters and almost empty ones. (see the discussion in
-    :ref:`sphx_glr_auto_examples_cluster_plot_agglomerative_clustering.py`).
+    :ref:`sphx_glr_auto_examples_cluster_plot_ward_structured_vs_unstructured.py`).
     Single linkage is the most brittle linkage option with regard to this issue.
 
-.. image:: ../auto_examples/cluster/images/sphx_glr_plot_agglomerative_clustering_001.png
-    :target: ../auto_examples/cluster/plot_agglomerative_clustering.html
+.. image:: ../auto_examples/cluster/images/sphx_glr_plot_ward_structured_vs_unstructured_001.png
+    :target: ../auto_examples/cluster/plot_ward_structured_vs_unstructured.html
     :scale: 38
 
-.. image:: ../auto_examples/cluster/images/sphx_glr_plot_agglomerative_clustering_002.png
-    :target: ../auto_examples/cluster/plot_agglomerative_clustering.html
-    :scale: 38
-
-.. image:: ../auto_examples/cluster/images/sphx_glr_plot_agglomerative_clustering_003.png
-    :target: ../auto_examples/cluster/plot_agglomerative_clustering.html
-    :scale: 38
-
-.. image:: ../auto_examples/cluster/images/sphx_glr_plot_agglomerative_clustering_004.png
-    :target: ../auto_examples/cluster/plot_agglomerative_clustering.html
+.. image:: ../auto_examples/cluster/images/sphx_glr_plot_ward_structured_vs_unstructured_002.png
+    :target: ../auto_examples/cluster/plot_ward_structured_vs_unstructured.html
```

Comment:
This part of the diff is sort of a regression in the docs.

Before the change, this is how it looks:

![image](https://github.com/user-attachments/assets/b8685c14-81f7-4a50-b288-6571844e83c1)

and after, it looks like this:

![image](https://github.com/user-attachments/assets/34a99a82-9651-42af-8cd5-af510bb4b3ee)

The "before" is much clearer and shows the intent of the warning very nicely. We should make sure the new example covers the topic and that the linked pictures also show the intent of the warning.